packages/ui/src/ui/containers/ManageTokens/ManageTokensPasswordModal/ManageTokensPasswordModal.tsx

```diff
@@ -41,26 +42,29 @@ const PasswordModal = (props: PasswordModalProps) => {
                     initialValues={{}}
                     onAdd={async (data) => {
                         setError(undefined);
-
-                        const password = data.getState().values.password;
-
-                        return sha256(password).then((password_sha256) => {
-                            axios
-                                .post(`/api/yt/${ytAuthCluster}/login`, {
-                                    username,
-                                    password,
-                                })
-                                .then(() => props.handleConfirm(password_sha256))
-                                .catch((error) => setError(error?.response?.data || error));
-                        });
+                        const {password, sha256_password} = data.getState().values;
+
+                        const strategy = createPasswordStrategy(
+                            username,
+                            ytAuthCluster,
+                            isSha256Mode(),
+                        );
+
+                        return strategy(sha256_password || password)
+                            .then((hashedPassword) => {
+                                props.handleConfirm(hashedPassword);
+                            })
+                            .catch((error: any) => setError(error?.response?.data || error));
                     }}
                     fields={[
                         {
                             name: 'error-block',
                             type: 'block',
                             extras: {
-                                children: (
+                                children: isSha256Mode() ? (
                                     <Alert message="To access tokens management, you need enter your password" />
+                                ) : (
+                                    <Alert message="For security purposes, we only accept SHA-256 hashed passwords. Please hash your password before entering it." />
```

Comment:
It is not about `security purposes`, it is about availability of `crypto.subtle`. It is better to mention it here:
```
crypto.subtle is not available from your browser, so SHA-256 hash cannot be generated. Please generate the hash of your password by self.
```
(`crypto.suptle` should be a link to `https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle`)